gdk4/src/gl_texture.rs

```diff
@@ -0,0 +1,44 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{GLContext, GLTexture, Texture};
+use glib::translate::*;
+
+impl GLTexture {
+    #[allow(clippy::new_ret_no_self)]
+    #[doc(alias = "gdk_gl_texture_new")]
+    pub unsafe fn new(context: &GLContext, id: u32, width: i32, height: i32) -> Texture {
```

Comment:
Shouldn't we return a `GLTexture` here by casting? Just like `gtk_button_new()` returns a `GtkWidget*` but `gtk::Button::new()` returns a `gtk::Button`.

Reply:
I don't mind, I was respecting the C API. But if in Rust we return the actual type, it's best to be consistent

Reply:
done

Reply:
Because you can directly use the API of all parent classes, that's what we do. In C the only reason why it is like that is because it's boring to write `GTK_WIDGET(button)` everywhere.